src/main/java/practice/CandidateValidator.java

```
@@ -1,5 +1,25 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate.getAge() >= 35
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals("Ukrainian")
+                && calculatePeriodsInUkr(candidate) >= 10;
+    }
+
+    private int calculatePeriodsInUkr(Candidate candidate) {
+        String periods = candidate.getPeriodsInUkr();
+        String[] years = periods.split("-");
+        if (years.length == 2) {
+            int startYear = Integer.parseInt(years[0]);
+            int endYear = Integer.parseInt(years[1]);
+            return endYear - startYear + 1;
+        }
+        return 0;
```

Returning 0 by default may hide errors in the input data. Consider throwing an exception or using an Optional to handle cases where the periods format is incorrect.